course/src/isometry.cc

```diff
@@ -312,74 +312,78 @@ namespace ekumen {
+    }
+    return Matrix3(
+        (row2_[1] * row3_[2] - row2_[2] * row3_[1]) / det,
+        (row2_[0] * row3_[2] - row2_[2] * row3_[0]) / det * -1,
+        (row2_[0] * row3_[1] - row2_[1] * row3_[0]) / det,
+        (row1_[1] * row3_[2] - row1_[2] * row3_[1]) / det * -1,
+        (row1_[0] * row3_[2] - row1_[2] * row3_[0]) / det,
+        (row1_[0] * row3_[1] - row1_[1] * row3_[0]) / det * -1,
+        (row1_[1] * row2_[2] - row1_[2] * row2_[1]) / det,
+        (row1_[0] * row2_[2] - row1_[2] * row2_[0]) / det * -1,
+        (row1_[0] * row2_[1] - row1_[1] * row2_[0]) / det
+      );
+  }
+
   // Isometry
 
   bool Isometry::operator == (const Isometry & obj) const
   {
-    return ((rotation_ == obj.rotation()) && (point_ == obj.point()));
+    return ((rotation_ == obj.rotation()) && (translation_ == obj.translation()));
   }
 
-  Vector3 Isometry::operator *= (const Vector3 & obj) const
+  Vector3 Isometry::operator * (const Vector3 & obj) const
   {
-    return Vector3(rotation_ * obj + point_);
+    return Vector3(rotation_ * obj + translation_);
   }
 
-  Isometry Isometry::operator *= (const Isometry & obj) const
+  Isometry Isometry::operator *= (const Isometry & obj)
   {
-    Matrix3 new_rotation = rotation_ * obj.rotation();
-    Vector3 new_point = rotation_ * obj.point() + point_;
-    return Isometry(new_point, new_rotation);
+    rotation_ *= obj.rotation();
+    translation_ = rotation_ * obj.translation() + translation_;
+    return *this;
   }
 
-  Vector3 Isometry::transform (const std::initializer_list<double> & obj) const
+  Vector3 Isometry::transform (const Vector3 & obj) const
   {
-    Vector3 temp(obj);
-    return Vector3(rotation_ * temp + point_);
+    return Vector3(rotation_ * obj + translation_);
   }
 
   Isometry Isometry::inverse () const
   {
-    double det = rotation_.det();
-    if (det == 0)
-    {
-      throw std::runtime_error("Matrix is not invertable");
-    }
-    Matrix3 new_rotation(
-        (rotation_[1][1] * rotation_[2][2] - rotation_[1][2] * rotation_[2][1]) / det,
-        (rotation_[1][0] * rotation_[2][2] - rotation_[1][2] * rotation_[2][0]) / det * -1,
-        (rotation_[1][0] * rotation_[2][1] - rotation_[1][1] * rotation_[2][0]) / det,
-        (rotation_[0][1] * rotation_[2][2] - rotation_[0][2] * rotation_[2][1]) / det * -1,
-        (rotation_[0][0] * rotation_[2][2] - rotation_[0][2] * rotation_[2][0]) / det,
-        (rotation_[0][0] * rotation_[2][1] - rotation_[0][1] * rotation_[2][0]) / det * -1,
-        (rotation_[0][1] * rotation_[1][2] - rotation_[0][2] * rotation_[1][1]) / det,
-        (rotation_[0][0] * rotation_[1][2] - rotation_[0][2] * rotation_[1][0]) / det * -1,
-        (rotation_[0][0] * rotation_[1][1] - rotation_[0][1] * rotation_[1][0]) / det
-      );
-    Vector3 new_point(new_rotation * point_);
-    return Isometry(-1 * new_point, new_rotation);
+    Matrix3 new_rotation = rotation_.inverse();
+    Vector3 new_translation(new_rotation * translation_);
```

Comment:
Prefer using the uniform initializer instead of assignment (as in 361) or  parenthesis (362). See https://isocpp.org/blog/2012/12/is-c11-uniform-initialization-a-replacement-for-the-old-style-syntax